### ISSUE TITLE
False positive for fn arg

### ISSUE THREAD

**Issue by wcarmon** [intellij-rust/intellij-rust#9844]
## Environment

* **IntelliJ Rust plugin version:**
- org.rust.lang (0.4.183.5026-222)

* **Rust toolchain version:**
- cargo 1.67.0-nightly (ba607b23d 2022-11-22)

* **IDE name and version:**
GoLand 2022.2.5
Build #GO-222.4459.23, built on November 21, 2022
Licensed to ...
Runtime version: 17.0.5+7-b469.71 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.4.0-128-generic
...
Current Desktop: XFCE

* **Operating system:**
* linux mint 5.4.0-128-generic

## Problem description
- false positive bug
- Rust plugin incorrectly marks fn arg usage as error

## Steps to reproduce
- put this into a rust file
- notice the error shown in screenshot (bug)
- attempt to compile with cargo build (succeeds)
```rust
fn a<F>(b: F) -> bool
where
    F: Fn(i32) -> bool,
{
    b(4)
}
```

![false-positive](https://user-images.githubusercontent.com/7406775/205509259-b7f0ee1d-206e-44c8-99b3-42c6d9a7a408.png)

**Comment by wcarmon** [comment]
Seems somewhat related to https://github.com/intellij-rust/intellij-rust/issues/9126

**Comment by dima74** [comment]
I think this should be fixed by #9761 (see https://github.com/intellij-rust/intellij-rust/issues/9748#issuecomment-1317312557 for details). The plugin version with the fix will be released tomorrow, alternatively, you can try nightly build ([instructions](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-quick-start.html#install-nightly))

**Comment by Undin** [comment]
Duplicate of #9748
